src/main/java/com/github/tomakehurst/wiremock/matching/NotPattern.java

```diff
@@ -0,0 +1,38 @@
+package com.github.tomakehurst.wiremock.matching;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+public class NotPattern extends StringValuePattern{
+
+    private StringValuePattern unexpectedPattern;
+
+    public NotPattern(@JsonProperty("not") StringValuePattern unexpectedPattern) {
+        super(unexpectedPattern.expectedValue);
+        this.unexpectedPattern = unexpectedPattern;
+    }
+
+    public StringValuePattern getNot(){
+        return unexpectedPattern;
+    }
+
+    @Override
+    public MatchResult match(String value) {
+        return invert(unexpectedPattern.match(value));
+    }
+
+    private MatchResult invert(final MatchResult matchResult){
+        return new MatchResult() {
+            @Override
+            public boolean isExactMatch() {
+                return !matchResult.isExactMatch();
+            }
+
+            @Override
+            public double getDistance() {
+                return 1.0 - matchResult.getDistance();
```

Comment:
I think this could result in the situation where `isExactMatch()` returns true but the distance is still > 0.

The logic here should probably be:
if (exact match)
  distance = 0;
else
  distance = 1.0

Reply:
Hello @tomakehurst we have made changes, sorry for commenting this twice, I commented the previous comment in overall thread, but I felt the comment belonged to this request thread.